xilem/examples/virtual_cats.rs

```diff
@@ -0,0 +1,201 @@
+use xilem_core::one_of::Either;
+
+/// The main state of the application.
+struct VirtualCats {
+    statuses: Vec<Status>,
+    tx: Option<Sender<u32>>,
+}
+
+#[derive(Debug)]
+struct Status {
+    code: u32,
+    message: &'static str,
+    image: ImageState,
+}
+
+#[derive(Debug)]
+/// The state of the download of an image from a URL
+enum ImageState {
+    NotRequested,
+    Pending,
+    // Error(?),
+    Available(Image),
+}
+
+impl VirtualCats {
+    fn virtual_item(&mut self, idx: i64) -> impl WidgetView<Self> + use<> {
+        let index: usize = idx.try_into().expect("VirtualScroll bounds set correctly.");
+        let item = self
+            .statuses
+            .get_mut(index)
+            .expect("VirtualScroll bounds set correctly.");
+        let img = match &item.image {
+            ImageState::NotRequested => {
+                // We can't store the items to be scheduled in `self`, because the parent view isn't re-run when this function is executed
+                // This is for the reasonable reason that that could create a loop, if this weren't conditional.
+                // However, we probably should have an escape hatch for when this is conditional.
+                self.tx.as_ref().unwrap().blocking_send(item.code).unwrap();
+                item.image = ImageState::Pending;
+                None
+            }
+            ImageState::Pending => None,
+            ImageState::Available(image) => Some(image),
+        };
+        let img = if let Some(img) = img {
+            let attribution = sized_box(
+                sized_box(
+                    prose("Copyright ©️ https://http.cat")
+                        .line_break_mode(LineBreaking::Clip)
+                        .alignment(TextAlignment::End),
+                )
+                .padding(4.)
+                .rounded(4.)
+                .background(palette::css::BLACK.multiply_alpha(0.5)),
+            )
+            .padding((30., 42., 0., 0.));
+            Either::A(zstack((
+                image(img).fit(ObjectFit::FitWidth),
+                attribution.alignment(Alignment::TopTrailing),
+            )))
+        } else {
+            Either::B(sized_box(spinner()).width(80.).height(80.))
+        };
+        flex((img, prose(item.message)))
+    }
+
+    fn view(&mut self) -> impl WidgetView<Self> + use<> {
+        let rx = if self.tx.is_none() {
+            let (tx, rx) = tokio::sync::mpsc::channel(10);
+            self.tx = Some(tx);
+            Some(Mutex::new(Some(rx)))
+        } else {
+            None
+        };
+        fork(
+            virtual_scroll(0..self.statuses.len() as i64, Self::virtual_item),
+            // TODO: This is really awful...
+            // Ultimately, this doesn't really make snse as a view?
+            task_raw(
+                move |proxy| {
+                    let mut rx = rx.as_ref().unwrap().lock().unwrap().take().unwrap();
```

Comment:
This is a messy line. I don't see a better alternative to it through without adding code elsewhere.